components/ai_chat/browser/engine/engine_consumer_llama.cc

```diff
@@ -0,0 +1,347 @@
+    SuggestedQuestionsCallback callback) {
+  std::string prompt;
+  std::vector<std::string> stop_sequences;
+  prompt = BuildLlama2GenerateQuestionsPrompt(is_video, page_content);
+  stop_sequences.push_back(kLlama2Eos);
+  stop_sequences.push_back("</ul>");
+  DCHECK(api_);
+  api_->QueryPrompt(
+      prompt, stop_sequences,
+      base::BindOnce(
+          &EngineConsumerLlamaRemote::OnGenerateQuestionSuggestionsResponse,
+          weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void EngineConsumerLlamaRemote::OnGenerateQuestionSuggestionsResponse(
+    SuggestedQuestionsCallback callback,
+    CompletionResult result) {
+  if (!result.has_value() || result->empty()) {
+    // Query resulted in error
+    LOG(ERROR) << "Error getting question suggestions.";
+    return;
+  }
+
+  // Success
+  // Llama 2 results look something like this:
+  // Can ChatGPT actually summarize a seven-hour video in under a minute?</li>
+  // <li>What are the limitations of ChatGPT's browsing capabilities, and how
+  // does it affect its ability to provide accurate information?</li> <li>Can
+  // ChatGPT's tonewood research be applied to other areas of scientific
+  // inquiry beyond guitar making?</li>  These questions capture interesting
+  // aspects of the video, such as the AI's ability to summarize long content,
+  // its limitations, and its potential applications beyond the specific use
+  // case presented in the video.
+
+  // Split out the questions using </li>
+  std::vector<std::string> questions = base::SplitStringUsingSubstr(
+      *result, "</li>", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+
+  // Remove the last entry in questions if it doesn't contain an <li> tag
+  // which means its not an actually a question
+  if (questions.size() > 1) {
+    if (questions.back().find("<li>") == std::string::npos) {
+      questions.pop_back();
+    }
+  }
+
+  // Remove leading <li> from each question
+  for (auto& question : questions) {
+    auto parts = base::SplitStringUsingSubstr(
+        question, "<li>", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+    if (!parts.empty()) {
+      question = parts.back();  // If there's an <li>, parts will contain an
+                                // empty string and then the question. We take
+                                // the last element.
+    }
+  }
+
+  std::move(callback).Run(std::move(questions));
+}
+
+void EngineConsumerLlamaRemote::SubmitHumanInput(
+    const bool& is_video,
+    const std::string& page_content,
+    const ConversationHistory& conversation_history,
+    const std::string& human_input,
+    CompletionDataReceivedCallback data_received_callback,
+    CompletionCompletedCallback completed_callback) {
+  std::string prompt = BuildLlama2Prompt(conversation_history, page_content,
+                                         is_video, human_input);
+  DCHECK(api_);
+  api_->QueryPrompt(prompt, {"</response>"}, std::move(completed_callback),
```

Comment:
@LorenzoMinto @nvonpentz should we be adding kLlama2Eos as a stop sequence here too, like we do with the question suggestions prompt?

Reply:
It should, we don't use that tag in llama. It might have gone overlooked because it's not explicit what it is. Could we maybe extract it to a named variable? A free string could be many different things without remembering the `QueryPrompt` signature

Reply:
done